hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxySupport.java

```diff
@@ -215,6 +233,23 @@ public boolean removeIfSameInternal(Data key, Object value) {
         return true;
     }
 
+
+    private void unlock(Data key, VersionedValue versionedValue) {
+        try {
+            NodeEngine nodeEngine = getNodeEngine();
+            TxnUnlockOperation unlockOperation = new TxnUnlockOperation(name, key, versionedValue.version);
+            unlockOperation.setThreadId(ThreadUtil.getThreadId());
+            unlockOperation.setOwnerUuid(tx.getOwnerUuid());
+            int partitionId = nodeEngine.getPartitionService().getPartitionId(key);
+            Future<VersionedValue> future = nodeEngine.getOperationService()
+                    .invokeOnPartition(MapService.SERVICE_NAME, unlockOperation, partitionId);
+            future.get();
+            valueMap.remove(key);
```

Comment:
The check (`valueMap.containsKey()`) is made to learn that whether lock is acquired by someone or I will be the one who will lock it first. If I'm the first one, I acquire the lock with `lockAndGet()` call which leads to a put call to the `valueMap`, so it is needed to remove the key when the key is unlocked.